sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala

```diff
@@ -463,7 +463,12 @@ class SparkSession private(
    */
   @Experimental
   def range(start: Long, end: Long, step: Long, numPartitions: Int): Dataset[java.lang.Long] = {
-    new Dataset(self, Range(start, end, step, numPartitions), Encoders.LONG)
+    val encoder = {
+      val schema = StructType(Seq(StructField("id", LongType, nullable = false)))
+      ExpressionEncoder[java.lang.Long]().copy[java.lang.Long](schema = schema)
+    }
+
+    new Dataset(self, Range(start, end, step, numPartitions), encoder)
```

Comment:
We are now using the encoder schema as Dataset schema, thus we need to rename the default primitive encoder column name "value" to the desired name.